contracts/0.4.24/oracle/OracleChiCaller.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: 2020 Lido <info@lido.fi>
+
+// SPDX-License-Identifier: GPL-3.0
+
+/* See contracts/COMPILERS.md */
+pragma solidity 0.4.24;
+
+import "../interfaces/ILidoOracle.sol";
+
+interface IFreeFromUpTo {
+    function freeFromUpTo(address from, uint256 value) external returns (uint256 freed);
+    function balanceOf(address account) external view returns (uint256);
+    function approve(address spender, uint256 amount) external returns (bool);
+}
+
+
+contract OracleChiCaller {
+    IFreeFromUpTo public constant chi = IFreeFromUpTo(0x0000000000004946c0e9F43F4Dee607b0eF1fA1c);
+    ILidoOracle public lidoOracle;
```

Comment:
needs to be permissioned (report should be onlyowner) otherwise everyone will be able to submit oracle reports.

one contract per oracle member

Reply:
Right, as discussed in private. It was the first draft, my main concern on this step was to calculate gas constants accurately.